### ISSUE TITLE
33.1.1.713 Android

### ISSUE THREAD

**Issue by urthling** [airsdk/Adobe-Runtime-Support#1569]
All is well on 698..

Seeing this when building 713 android..

```
Error occurred while installing the application:
adb: failed to install /path/myapk: Failure [INSTALL_PARSE_FAILED_UNEXPECTED_EXCEPTION: Failed to parse /data/app/vmdl1116335577.tmp/base.apk: AndroidManifest.xml]
Performing Streamed Install
```

**Comment by ajwfrost** [comment]
This is a strange one: were you packaging as AAB or as APK? and are you able to check that it's not an App Bundle within an APK filename? (if you open the package in a zip manager and look at the top-level contents: AAB has a "BundleConfig.pb" file, APK has the AndroidManifest.xml file there).

If you can also open your package using Android Studio (from the build menu to analyze an APK/Bundle) and let us know what's in the AndroidManifest.xml file, we can see if there's anything that's got malformed..

thanks

**Comment by leossmith** [comment]
I get the same error when using .aab. It installs when using apk but doesn't open it for debug (i had to manually run the app)

**Comment by ajwfrost** [comment]
@leossmith can you clarify please: if you package as .aab, then you are also seeing that it doesn't install?
For .apk, you can install it but not run it?
We did have an issue here:
#1430 : ADT failing to launch an Android application when using new build mechanism
But this was fixed in 33.1.1.713...

Failing to install an .aab file is a new one that we've not seen and currently can't reproduce, so I'm wondering whether there's something in the XML descriptor that is resulting in a malformed AndroidManifest.xml file..

**Comment by leossmith** [comment]
@ajwfrost exactly. When I set .aab in adt.cfg then it doesnt even install on my device during debug. 

When I change to .apk it installs but I get an error that it couldn't open the app. However if I run the installed app on my own, debug starts / connects.

As I am always using a lot of ANEs this might be causing some issues with AndroidManifest. I will replicate the issue later today, so I can post all the errors I get.

**Comment by ajwfrost** [comment]
Thanks @leossmith ...
So the change to .apk and failure to open looks like the issue with launching legacy-built APKs that was just mentioned in #1430, so yes we had a bug in that where it didn't pick up the installation failure with the 'new' activity name, it was meant to try again with the 'old' activity name but that part was missing.

But still a bit concerning if the AAB install fails. Wondering if you can try it in two steps, and/or inspect the bundle in Android Studio to extract the manifest XML?
```
java -jar bundletool.jar build-apks --bundle output.aab --output output.apks --connected-device
java -jar bundletool.jar install-apks --apks=output.apks
```
thanks

**Comment by leossmith** [comment]
This is what I get trying to debug with aab (app never installs):
![image](https://user-images.githubusercontent.com/4300876/149771035-a8f66334-34b2-43fc-8f47-5eef4375b3cc.png)

This is what I get trying to debug with apk (app installs but i have to run it):
![image](https://user-images.githubusercontent.com/4300876/149773205-921382f1-6ce9-4501-adc8-4faf6bfe7b57.png)

to get here I have to disable both aab and android sdk in adt.cfg and use buildLegacyAPK

If I try with buildlegacy false then I get :
![image](https://user-images.githubusercontent.com/4300876/149771761-13b5a675-cc1f-45cf-9b0b-29b2362ba5df.png)

Using the java commands you sent, work to install the app on the connected device.

**Comment by ajwfrost** [comment]
Okay thanks .. 
For the first of these ("install android application" error message), it's odd that this fails from ADT but works when you use the raw Java commands. Could you please try taking the updated ADT.jar file from the below link, and setting up your config to have "DebugOut=true" - and then post the ~/adt.log file contents please?
https://transfer.harman.com/message/yceO1CHRhb16z9lif4pObq

thanks

**Comment by urthling** [comment]
Awesome, that update did it for me -- I can now build and install using .713.  I'm manually opening the App though but no build failures.  I can also create a production build with `CreateAndroidAppBundle=true`

**Comment by leossmith** [comment]
I am still getting the same behavior as before, after replacing adt.jar with the one provided above.

**Comment by leossmith** [comment]
This is what i get:

```*** ADT Logging at 2022/01/18 15:17:56 ***
Called by: I:\Development\flex-sdks\4161_331_633\lib\adt.jar
ADT 33.1.1.743 called with: -installApp -platform android -device 45768874 -package E:/Development/Bitbucket/trekrecord-flex/bin-debug/Main.apk
Running ADB command line: I:\Development\android\sdk\platform-tools\adb.exe -s 45768874 devices
ADB output = List of devices attached
45768874	device

Running ADB command line: I:\Development\android\sdk\platform-tools\adb.exe -s 45768874 install E:\Development\Bitbucket\trekrecord-flex\bin-debug\Main.apk
ADB output = Performing Streamed Install
ADB error  = adb: failed to install E:\Development\Bitbucket\trekrecord-flex\bin-debug\Main.apk: Failure [INSTALL_PARSE_FAILED_UNEXPECTED_EXCEPTION: Failed to parse /data/app/vmdl322734239.tmp/base.apk: AndroidManifest.xml]
..java.base/java.lang.Thread.getStackTrace(Thread.java:1602)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.android.AndroidDeviceOperations._execADB(AndroidDeviceOperations.java)
..com.adobe.air.android.AndroidDeviceOperations.execADB(AndroidDeviceOperations.java)
..com.adobe.air.android.AndroidDeviceOperations.installAPK(AndroidDeviceOperations.java)
..com.adobe.air.android.AndroidDeviceOperations.installApplication(AndroidDeviceOperations.java)
..com.adobe.air.ADT.parsePlatformOperationArgs(ADT.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
..java.base/java.lang.Thread.getStackTrace(Thread.java:1602)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.android.AndroidDeviceOperations.installAPK(AndroidDeviceOperations.java)
..com.adobe.air.android.AndroidDeviceOperations.installApplication(AndroidDeviceOperations.java)
..com.adobe.air.ADT.parsePlatformOperationArgs(ADT.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
ADT exception: Failure [INSTALL_PARSE_FAILED_UNEXPECTED_EXCEPTION: Failed to parse /data/app/vmdl322734239.tmp/base.apk: AndroidManifest.xml]

```

**Comment by ajwfrost** [comment]
Okay thanks @leossmith  .. it seems that's all the information that ADB is giving us. Could you try it manually from a command line, and add some verbosity to the output? i.e.
```
set ADB_TRACE=all
I:\Development\android\sdk\platform-tools\adb.exe -s 45768874 install E:\Development\Bitbucket\trekrecord-flex\bin-debug\Main.apk
```

Plus if you can share your APK with us we can try it here and see if there are issues. Just wondering, given some of the other folk online with this issue, whether it's only happening on certain versions of Android on the target device..?

thanks
